mlflow_faculty/trackingstore.py

```diff
@@ -189,10 +190,26 @@ def update_run_info(self, run_uuid, run_status, end_time):
         """
         Updates the metadata of the specified run.
 
+        :param experiment_id: ID of the experiment for this run
+        :param user_id: ID of the user launching this run
+        :param source_type: Enum (integer) describing the source of the run
+        :param tags: List of Mlflow Tag entities.
+
         :return: :py:class:`mlflow.entities.RunInfo` describing the updated
             run.
         """
-        raise NotImplementedError()
+        try:
+            faculty_run = self._client.update_run_info(
+                self._project_id,
+                UUID(run_uuid),
+                mlflow_to_faculty_run_status(run_status),
+                end_time,
+            )
+        except faculty.clients.base.HttpError as e:
+            raise faculty_http_error_to_mlflow_exception(e)
+        else:
+            mlflow_run = faculty_run_to_mlflow_run(faculty_run)
+            return mlflow_run
```

Comment:
I think this should just return the run info, not the full run?

https://github.com/mlflow/mlflow/blob/1b03acca64a0a6ead65d7fd8e54222804d90f9e1/mlflow/store/abstract_store.py#L122